pkg/proxy/endpoints_test.go

```diff
@@ -54,15 +54,15 @@ func TestGetLocalEndpointIPs(t *testing.T) {
 		// Case[1]: unnamed port
 		endpointsMap: EndpointsMap{
 			makeServicePortName("ns1", "ep1", "", v1.ProtocolTCP): []Endpoint{
-				&BaseEndpointInfo{Endpoint: "1.1.1.1:11", IsLocal: false, Ready: true, Serving: true, Terminating: false},
+				NewTestEp("1.1.1.1", 11, "", false, true, true, false),
```

Comment:
Meh. No, this is no good; there's no way anyone could look at "false, true, true, false" and understand that this was a non-local serving-non-terminating endpoint. :confounded: 

(`NewBaseEndpointInfo` doesn't have the same problem because the only place that calls it passes variables rather than constants, so it would look like `..., local, ready, serving, terminating)` rather than `..., false, true, true, false)`.)

I'm not sure what the best fix is now... I thought about just changing the unit tests to set all three fields instead (`Endpoint`, `ip`, and `port`), but that requires renaming `ip` and `port` to `IP` and `Port` so that `pkg/proxy/ipvs` can set them too, which then requires renaming `Endpoint.IP()` and `Endpoint.Port()` to `GetIP()` and `GetPort()`... You can see the result of that here: https://github.com/danwinship/kubernetes/commits/pr120104

@aojea what do you think? It seems like all possibilities are going to be a little bit ugly, due to the current state of the APIs and the unit tests...

Reply:
I have also considered changing `ip` and `port` to `IP` and `Port`, but as you mentioned, it would require modifying `Endpoint.IP()` and `Endpoint.Port()`.

As for "false, true, true, false", I use Goland to write code, which automatically suggests parameter names. So, I find it feasible, as shown in the screenshot:

<img width="1693" alt="image" src="https://github.com/kubernetes/kubernetes/assets/55381228/321ad3ce-7892-4e6a-8ec1-01f08b59bb14">

Reply:
https://github.com/danwinship/kubernetes/commits/pr120104

thanks @danwinship 

Changing `ip` and `port` to `IP` and `Port`, `IP()` and `Port()` to `GetIP()` and `GetPort()`, does seem to make it better.

Because my current `NewTestEp` method is placed in the `endpoints.go`, it's not very good.

Reply:
> @aojea what do you think? It seems like all possibilities are going to be a little bit ugly, due to the current state of the APIs and the unit tests...

I came to the same conclusion ... we have to live with our mistakes :/

Reply:
@togettoyou ok, can you re-push your branch with the commits from my branch then?

Reply:
actually, I just found that in one of my refactoring branches I have a better cleanup for those unit tests... let me poke around at this again and I'll push a PR and either include your patch, or else you can rebase your branch on mine after it merges...

Reply:
#121097

Reply:
> actually, I just found that in one of my refactoring branches I have a better cleanup for those unit tests... let me poke around at this again and I'll push a PR and either include your patch, or else you can rebase your branch on mine after it merges...

ok, thank you. I will continue to follow up after your branch is successfully merged.